Fixes incorrect desktop video resolution when using HiDPI

initramfs opened the pull request:
When attempting to use "Desktop" as the video source under HiDPI scaling, the resolution returned by Qt's `QApplication::desktop()->screenGeometry().size()` would return a resolution in device-indepedent pixels (which would be less than the actual physical number of pixels being projected by the monitor). As such, toxav ends up clipping the desktop feed to only the given region, resulting in a subset of the desktop being displayed. This issue is examplified below:
![clipped](https://cloud.githubusercontent.com/assets/3128050/14221679/b4311918-f836-11e5-8cb0-7832b43d683a.png)

The problem is resolved by usage of the function `QScreen::devicePixelRatio()` which returns the scaling factor between the DPI pixels and the actual pixel pixels.

The one caveat with ths solution is that `devicePixelRatio()` is only available >= Qt 5.5. A temporary solution is to assume a default of no scaling when compiling with Qt < 5.5.0 (matches current behaviour).

initramfs committed:
fix(video): incorrect desktop video resolution when using HiDPI

Fixes desktop sharing region being clipped when DPI scaling is being
used.

GitCop commented:
There were the following issues with your Pull Request
- Commit: d3f3dca045628c75687e0a8bba3e5c573a13eb87
  - Invalid type. Valid types are **feat,  fix,  docs,  style,  refactor,  perf,  test,  chore,  revert**

Guidelines are available at
https://github.com/tux3/qTox/blob/master/CONTRIBUTING.md#commit-message-format

If this message was triggered by missing `(<scope>)` simply ignore it.

---

This message was auto-generated by https://gitcop.com

zetok commented:
Huh. Damn, looks like gitcop is misconfigured, or is failing to do what it is supposed to do. Looks like it can't be used after all. :/

tux3 commented:
Maybe gitcop is confused, I noticed that by default gitcop is configured to allow a maximum of 72 chars in the subject AND in every line of the body, and this commit has a 75 character line.
So I upped the limit in the message body to 255, let's see if it helps.

initramfs commented:
@tux3 According to the [contributing guidelines](https://github.com/tux3/qTox/blob/master/CONTRIBUTING.md#commit-message-format), shouldn't the maximum commit line length be limited to 100 (and not 72)? Wouldn't a limit of 100 be sufficient?

zetok commented:
@initramfs right, sorry about that.

Related: https://github.com/tux3/qTox/issues/3089#issuecomment-204174033

GitCop commented:
There were the following issues with your Pull Request
- Commit: a54601f072c9dd277cb757573caa5dc99d4576bb
  - Invalid type. Valid types are **feat,  fix,  docs,  style,  refactor,  perf,  test,  chore,  revert**

Guidelines are available at
https://github.com/tux3/qTox/blob/master/CONTRIBUTING.md#commit-message-format

If this message was triggered by missing `(<scope>)` simply ignore it.

---

This message was auto-generated by https://gitcop.com

initramfs commented:
Updated commit message to fix spelling mistake.

tux3 commented:
Welp, I'll just disable GitCop.

sudden6 commented on src/video/cameradevice.cpp:
```diff
@@ -132,11 +133,20 @@ CameraDevice* CameraDevice::open(QString devName, VideoMode mode)
         else
         {
-            screen = QApplication::desktop()->screenGeometry().size();
+            QScreen* defaultScreen = QApplication::screens().at(0);
```
Will this also work in a multi-monitor environment?
Shouldn't https://doc.qt.io/qt-5/qguiapplication.html#primaryScreen-prop be used?

initramfs committed:
fix(video): uses explicit default screen from QGuiApplication

This commit changes the method used to obtain the active screen from
simply selecting the first screen in the list to using the explicit
default screen from QGuiApplication::primaryScreen() function

initramfs commented:
@sudden6 Sorry I didn't know about the existance of that function, I've added a commit to change the behaviour to use the aforementioned function, it should now explictily use the default screen. 

Regarding support in a multimonitor environment, before the latest commit, you are right in that it could accidentally select the second monitor if that monitor was somehow index 0. However, I believe a much better solution would be to give users a selector to show which monitor to display.

zetok commented:
> However, I believe a much better solution would be to give users a selector to show which monitor to display.

Yeah, until that will be possible though, selecting default screen is probably the best :)

zetok commented:
RE: screen selection:
Related: #1786

sudden6 commented:
@initramfs if you are going to add a way to choose the screen please open a new PR.
